play-json-jsoniter/shared/src/main/scala/com/evolutiongaming/jsonitertool/JsonValueCodecJsValue.scala

```diff
@@ -75,8 +75,8 @@ object JsonValueCodecJsValue {
         }
       }
 
-      def encodeValue(jsValue: JsValue, out: JsonWriter): Unit = {
-        jsValue match {
+      def encodeValue(jsValue: JsValue, out: JsonWriter): Unit =
+        (jsValue: @unchecked) match {
```

Comment:
why unchecked here?

Reply:
Because the compiler warns "match is not exhaustive".

Reply:
should we make exhaustive and fail on unsupported cases?

Reply:
Missing cases are redundant `JsTrue` and `JsFalse` values.

Reply:
I understand that we are talking here about closed options of enums, but in general usually it's better to unmatch all cases so we would get a warn just in case of new child is coming from library version bump

Reply:
I've added a default case for all unsupported yet types of values.